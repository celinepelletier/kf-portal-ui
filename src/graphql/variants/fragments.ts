import { gql } from '@apollo/client';

export const genesFragment = gql`
  fragment genesFragment on variantsGenes {
    hits {
      total
      edges {
        node {
          score
          ensembl_gene_id
          entrez_gene_id
          hgnc
          location
          name
          omim_gene_id
          symbol
          alias

          cosmic {
            hits {
              total
              edges {
                node {
                  id
                  score
                  tumour_types_germline
                }
              }
            }
          }
          ddd {
            hits {
              total
              edges {
                node {
                  id
                  score
                  disease_name
                }
              }
            }
          }
          hpo {
            hits {
              total
              edges {
                node {
                  score
                  hpo_term_id
                  hpo_term_label
                  hpo_term_name
                }
              }
            }
          }
          omim {
            hits {
              total
              edges {
                node {
                  score
                  inheritance
                  inheritance_code
                  name
                  omim_id
                }
              }
            }
          }
          orphanet {
            hits {
              total
              edges {
                node {
                  score
                  inheritance
                  disorder_id
                  panel
                }
              }
            }
          }
        }
      }
    }
  }
`;

export const frequenciesFragment = gql`
  fragment frequenciesFragment on variantsFrequencies {
    gnomad_exomes_2_1 {
      ac
      af
      an
      homozygotes
    }
    gnomad_genomes_2_1 {
      ac
      af
      an
      homozygotes
    }
    gnomad_genomes_3_0 {
      ac
      af
      an
      homozygotes
    }
    gnomad_genomes_3_1_1 {
      ac
      af
      an
    }
    one_thousand_genomes {
      ac
      af
      an
    }
    topmed {
      ac
      af
      an
      homozygotes
      heterozygotes
    }
    internal {
      lower_bound_kf {
        ac
        af
        an
        heterozygotes
        homozygotes
      }
      upper_bound_kf {
        ac
        af
        an
        heterozygotes
        homozygotes
      }
    }
  }
`;

export const consequencesFragment = gql`
  fragment consequencesFragment on variantsConsequences {
    hits {
      total
      edges {
        node {
          predictions {
            fathmm_pred
            lrt_pred
            lrt_converted_rankscore
            revel_rankscore
            sift_pred
            polyphen2_hvar_pred
            polyphen2_hvar_rankscore
            sift_converted_rankscore
            cadd_rankscore
            dann_rankscore
            fathmm_converted_rankscore
          }
          hgvsc
          consequences
        }
      }
    }
  }
`;

export const clinvarFragment = gql`
  fragment clinvarFragment on variantsClinvar {
    clin_sig
    clinvar_id
    conditions
    inheritance
    interpretations
  }
`;

export const studiesVariantFragment = gql`
  fragment studiesVariantFragment on variantsStudies {
    hits {
      total
      edges {
        node {
          score
          participant_ids
          participant_number
          study_code
          study_id
          acls
          external_study_ids
          transmissions
          frequencies {
            lower_bound_kf {
              ac
              af
              an
              heterozygotes
              homozygotes
            }
            upper_bound_kf {
              ac
              af
              an
              heterozygotes
              homozygotes
            }
          }
        }
      }
    }
  }
`;
