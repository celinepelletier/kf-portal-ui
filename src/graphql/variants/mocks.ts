import { IVariantEntity as IVariantEntityFerlab } from '@ferlab/ui/core/pages/EntityPage/type';

export const variantDataMock: IVariantEntityFerlab = {
  id: 'T_K63HwB7ARyLJRSUgGt',
  alternate: 'G',
  chromosome: '10',
  genome_build: 'GRCh38',
  hgvsg: 'chr10:g.123137309del',
  locus: '10-123137306-GA-G',
  participant_number: 1,
  participant_total_number: 11030,
  reference: 'GA',
  rsnumber: '',
  start: 123137306,
  variant_class: 'deletion',
  studies: {
    hits: {
      total: 1,
      edges: [
        {
          node: {
            id: 'test',
            participant_ids: [],
            participant_number: 1,
            study_id: 'SD_DYPMEHHF',
            study_code: 'KF-NBL',
            frequencies: {
              upper_bound_kf: {
                ac: 1,
                homozygotes: 0,
              },
            },
          },
        },
      ],
    },
  },
  consequences: {
    hits: {
      total: 1,
      edges: [
        {
          node: {
            symbol: 'HMX3',
            biotype: 'protein_coding',
            consequences: ['frameshift_variant'],
            //@ts-ignore type is wrong, value is valie from BE
            vep_impact: 'HIGH',
            impact_score: 4,
            canonical: true,
            strand: '',
            refseq_mrna_id: 'NM_001105574',
            ensembl_transcript_id: 'ENST00000357878',
            ensembl_gene_id: 'ENSG00000188620',
            hgvsc: 'ENST00000357878.5:c.652del',
            hgvsp: 'ENSP00000350549.4:p.Ser218ValfsTer31',
            predictions: {
              fathmm_pred: 0,
              lrt_pred: '',
              lrt_converted_rankscore: 0,
              revel_rankscore: 0,
              sift_pred: '',
              polyphen2_hvar_pred: '',
              sift_converted_rankscore: 0,
              cadd_rankscore: 0,
              dann_rankscore: 0,
              fathmm_converted_rankscore: 0,
              polyphen2_hvar_rankscore: 0,
            },
            conservations: {
              phylo_p17way_primate_rankscore: 0,
            },
          },
        },
      ],
    },
  },
  //@ts-ignore
  clinvar: '',
  frequencies: {
    internal: {
      upper_bound_kf: {
        ac: 1,
        homozygotes: 0,
      },
    },
    topmed: {
      ac: 0,
      af: 0,
      an: 0,
      homozygotes: 0,
    },
    one_thousand_genomes: {
      ac: 0,
      af: 0,
      an: 0,
      homozygotes: 0,
    },
    gnomad_exomes_2_1: {
      ac: 0,
      af: 0,
      an: 0,
      homozygotes: 0,
    },
    gnomad_genomes_2_1: {
      ac: 0,
      af: 0,
      an: 0,
      homozygotes: 0,
    },
    gnomad_genomes_3_0: {
      ac: 0,
      af: 0,
      an: 0,
      homozygotes: 0,
    },
    gnomad_genomes_3_1_1: {
      ac: 0,
      af: 0,
      an: 0,
      homozygotes: 0,
    },
  },
  genes: {
    hits: {
      total: 1,
      edges: [
        {
          node: {
            location: '10q26.13',
            omim_gene_id: '613380',
            symbol: 'HMX3',
            cosmic: {
              hits: {
                total: 0,
                edges: [],
              },
            },
            ddd: {
              hits: {
                total: 0,
                edges: [],
              },
            },
            hpo: {
              hits: {
                total: 0,
                edges: [],
              },
            },
            omim: {
              hits: {
                total: 0,
                edges: [],
              },
            },
            orphanet: {
              hits: {
                total: 0,
                edges: [],
              },
            },
            id: undefined,
          },
        },
      ],
    },
  },
};
