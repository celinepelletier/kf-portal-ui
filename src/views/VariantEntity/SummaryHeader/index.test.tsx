import { StaticRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { IVariantEntity as IVariantEntityFerlab } from '@ferlab/ui/core/pages/EntityPage/type';
import { variantDataMock } from 'graphql/variants/mocks';

import SummaryHeader from './index';

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  // 3- Import non-mocked library and use other functionalities and hooks
  ...(jest.requireActual('react-router-dom') as any),

  // 4- Mock the required hook
  useNavigate: () => mockedUsedNavigate,
}));

jest.mock('react-intl-universal', () => ({ get: jest.fn() }));

describe('SummaryHeader', () => {
  let data: IVariantEntityFerlab;

  beforeEach(() => {
    data = variantDataMock;
  });

  it('renders correctly', () => {
    //@ts-ignore
    const tree = renderer.create(
      <StaticRouter location="someLocation">
        <SummaryHeader variant={data} />
      </StaticRouter>,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
