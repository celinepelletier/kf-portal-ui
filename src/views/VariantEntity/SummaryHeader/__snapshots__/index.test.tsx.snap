// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryHeader renders correctly 1`] = `
<div
  className="container"
>
  <a
    className="link"
    href="/data-exploration/participants"
    onClick={[Function]}
  >
    <span
      aria-label="read"
      className="anticon anticon-read icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="read"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M928 161H699.2c-49.1 0-97.1 14.1-138.4 40.7L512 233l-48.8-31.3A255.2 255.2 0 00324.8 161H96c-17.7 0-32 14.3-32 32v568c0 17.7 14.3 32 32 32h228.8c49.1 0 97.1 14.1 138.4 40.7l44.4 28.6c1.3.8 2.8 1.3 4.3 1.3s3-.4 4.3-1.3l44.4-28.6C602 807.1 650.1 793 699.2 793H928c17.7 0 32-14.3 32-32V193c0-17.7-14.3-32-32-32zM324.8 721H136V233h188.8c35.4 0 69.8 10.1 99.5 29.2l48.8 31.3 6.9 4.5v462c-47.6-25.6-100.8-39-155.2-39zm563.2 0H699.2c-54.4 0-107.6 13.4-155.2 39V298l6.9-4.5 48.8-31.3c29.7-19.1 64.1-29.2 99.5-29.2H888v488zM396.9 361H211.1c-3.9 0-7.1 3.4-7.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c.1-4.1-3.1-7.5-7-7.5zm223.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c0-4.1-3.2-7.5-7.1-7.5H627.1c-3.9 0-7.1 3.4-7.1 7.5zM396.9 501H211.1c-3.9 0-7.1 3.4-7.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c.1-4.1-3.1-7.5-7-7.5zm416 0H627.1c-3.9 0-7.1 3.4-7.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c.1-4.1-3.1-7.5-7-7.5z"
        />
      </svg>
    </span>
    <span
      className="entityCount"
    >
      1
    </span>
    <span
      className="text"
    />
  </a>
  <div
    className="participantDisabled"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      aria-label="user"
      className="anticon anticon-user icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="user"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
        />
      </svg>
    </span>
    <span
      className="entityCount"
    >
      1
    </span>
    <span
      className="text"
    />
  </div>
</div>
`;
